keps/sig-architecture/5241-beta-featuregate-promotion-requirements/README.md

```diff
@@ -0,0 +1,246 @@
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Features gates must include all functional, security, and testing requirements along with
+resolving all issues and gaps identified prior to being enabled by default.
+The only valid GA criteria are “all issues and gaps identified as feedback during beta are resolved”.
+
+## Motivation
+
+Features gates that are enabled by default are enabled in every production Kubernetes cluster in the world.
+We must avoid making every production cluster into unstable or incomplete feature testing clusters.
+Even feature gates that make flags accessible, but require a secondary configuration to use must be
+stable, because it is unrealistic to expect everyone to understand the graduation stages of various flags
+for each release: the only stages that really matter are "takes enabling an explicit alpha feature gate"
+and "my production cluster accepts this as valid by default".
+
+### Goals
+
+* Features gates must include all functional, security, and testing requirements along with
+  resolving all issues and gaps identified prior to being enabled by default.
+* The only valid GA criteria are “all issues and gaps identified as feedback during beta are resolved”.
+
+### Non-Goals
+
+* Changing beta APIs off by default rules.
+* Change the imperfect mechanisms we have for API evolution.
+
+## Proposal
+
+Kubernetes feature gates have three levels: GA, Beta, and Alpha.
+1. GA means that a feature gate is unconditionally enabled in all production kubernetes clusters and
```

Comment:
we have two levels here GA, and GA locked by default, without locked by default it can be disabled, right?